cmd/gen-go-sample/inittree.go

```diff
@@ -56,37 +58,65 @@ type initType struct {
 	valFmt func(*generator, string) (string, error)
 }
 
+func (t *initType) isMap() (bool, error) {
+	if t.keyType != nil && t.valueType != nil {
+		return true, nil
+	}
+	if t.keyType == nil && t.valueType == nil {
+		return false, nil
+	}
+	return false, errors.E(nil, "internal: only one of keyType and valueType is set.")
+}
+
 // initTree represents a node in the initialization tree.
 type initTree struct {
 	typ initType
 
 	// initTree is either a composite value (struct/array/map), where keys and vals are set,
 	// or a simple value, where leafVal is set.
 
+	// vals and keys keep tracks of all the child trees of this inittree and their identifiers.
+	// If the child is a protobuf field, key is the field name.
+	// If the child is an array element, key is the index.
```

Comment:
Just so I understand: for a repeated proto field, there's an `initTree` with the index, and the child has a key with the field name? Or vice-versa?

Reply:
> for a repeated proto field, there's an initTree with the index, and the child has a key with the field name?
Yes!